### PR TITLE
Bring up back

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: cli/pkg/localbuilder/builder.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+type LocalBuilder interface {
+	Setup() error
+	Build(map[string]riov1.ImageBuild, bool, string) (map[string]string, error)
+}
+
+type RuntimeBuilder interface {
+	Build(riov1.ImageBuild) (string, error)
+}
+
+func NewLocalBuilder(ctx context.Context, apply apply.Apply, k8s *kubernetes.Clientset) (LocalBuilder, error) {
```
Comment (ibuildthecloud) [inline cli/pkg/localbuilder/builder.go]: do Setup() here.

File: cli/pkg/localbuilder/builder.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	dockerSocket      = "/run/docker.sock"
+	containerdSocket  = "/run/k3s/containerd/containerd.sock"
+	buildkitNamespace = "default"
+
+	maxBuildThread = int64(10)
+)
+
+type LocalBuilder interface {
+	Setup() error
+	Build(map[string]riov1.ImageBuild, bool, string) (map[string]string, error)
```
Comment (ibuildthecloud) [inline cli/pkg/localbuilder/builder.go]: add ctx args

Comment (ibuildthecloud) [reply cli/pkg/localbuilder/builder.go]: and drop Setup from interface

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rio/issues/346

File: cli/pkg/localbuilder/builder.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	apply          apply.Apply
+	k8s            *kubernetes.Clientset
+	runtime        string
+	runtimeBuilder RuntimeBuilder
+	buildkitPort   string
+	socketPort     string
+}
+
+func (l localBuilder) Build(ctx context.Context, specs map[string]riov1.ImageBuild, parallel bool, namespace string) (map[string]string, error) {
+	if err := l.setup(ctx); err != nil {
```
Comment (ibuildthecloud) [inline cli/pkg/localbuilder/builder.go]: move to new

File: cli/pkg/localbuilder/builder.go [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+	buildkitNamespace = "default"
+
+	maxBuildThread = int64(10)
+
+	base = 32768
+	end  = 61000
+)
+
+type LocalBuilder interface {
+	Build(context.Context, map[string]riov1.ImageBuild, bool, string) (map[string]string, error)
```
Comment (ibuildthecloud) [inline cli/pkg/localbuilder/builder.go]: arguement names

File: cli/pkg/localbuilder/builder.go [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+}
+
+type RuntimeBuilder interface {
+	Build(context.Context, riov1.ImageBuild) (string, error)
```
Comment (ibuildthecloud) [inline cli/pkg/localbuilder/builder.go]: argument names

File: pkg/stack/stack.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	}
+
+	if err := t.Validate(); err != nil {
+		return nil, err
+	}
+
+	return t.Questions()
+}
+
+func (s *Stack) content() ([]byte, error) {
```
Comment (ibuildthecloud) [inline pkg/stack/stack.go]: this is system only logic

Comment (ibuildthecloud) [reply pkg/stack/stack.go]: maybe?

File: pkg/stack/stack.go [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+	}
+
+	rf, err := riofile.Parse(bytes.NewBuffer(content), template.AnswersFromMap(s.answers))
+	if err != nil {
+		return nil, err
+	}
+	return rf.Objects(), nil
+}
+
+func (s *Stack) SetObjects(objects []runtime.Object) *Stack {
```
Comment (ibuildthecloud) [inline pkg/stack/stack.go]: WithObjects

File: pkg/stack/stack.go [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+		if svc, ok := obj.(*riov1.Service); ok {
+			if svc.Spec.Image == "" {
+				buildConfig[svc.Name] = *svc.Spec.Build
+			}
+		}
+	}
+	return buildConfig, nil
+}
+
+func (s *Stack) SetImageBuilds(images map[string]string) error {
```
Comment (ibuildthecloud) [inline pkg/stack/stack.go]: SetServiceImages

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rio/issues/347
https://github.com/rancher/rio/issues/346

**Pull request merged**